homeassistant/components/rainbird.py

```diff
@@ -0,0 +1,49 @@
+"""
+Support for Rain Bird Irrigation system LNK WiFi Module.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/rainbird/
+"""
+
+import logging
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.const import (CONF_HOST, CONF_PASSWORD)
+
+REQUIREMENTS = ['pyrainbird==0.1.2']
+
+DATA_RAINBIRD = 'rainbird'
+DOMAIN = 'rainbird'
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up Rain Bird componenent."""
+    conf = config[DOMAIN]
+    server = conf.get(CONF_HOST)
+    password = conf.get(CONF_PASSWORD)
+
+    from pyrainbird import RainbirdController
+    controller = RainbirdController(_LOGGER)
```

Comment:
Why pass the logger instance? The library should use its own logger.

Reply:
It's customary that each module should implement a logger like this:
```py
_LOGGER = logging.getLogger(__name__)
```
That way the origin of the log will be clear.

Reply:
Fixed